### PR TITLE
feat(msk): add Kafka version 3.3.2

### LINKED ISSUE #24432: feat(msk): add Kafka versions 3.3.2
### Describe the feature

Add support for Apache Kafka versions 3.3.2 in Amazon MSK.

Announcement:

[3.3.2](https://aws.amazon.com/de/about-aws/whats-new/2023/03/amazon-msk-apache-kafka-version-3-3-2/) (Posted On: Mar 02, 2023)

### Use Case

Being able to use the latest patch version of the MSK offering.

### Proposed Solution

Update of the supported versions in the class KafkaVersion.
(similar to [https://github.com/aws/aws-cdk/issues/23899](https://github.com/aws/aws-cdk/issues/23899))

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.67.0

### Environment details (OS name and version, etc.)

Ubuntu 22.04

### PR DESCRIPTION
**Pull request by stefanfreitag**
Add support for the MSK Kafka version 3.3.2.
The new version was announced on Mar 02 2023, the link to the announcement is in the linked issue.

Closes #24432 .

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by stefanfreitag** [comment]
Hi,
in my opinion this change does not require an update of the README.md file. It introduces a new entry in the KafkaVersions class and those entries are not documented in the README.
Please let me know if you are fine with this.

**Review by kaizencc — APPROVED** [review]

**Comment by stefanfreitag** [comment]
@Mergifyio requeue

**Comment by stefanfreitag** [comment]
@kaizencc Is it possible to re-trigger the code build job from your side?  I had seen 

```
[2/4] Fetching packages...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.npmjs.org/typescript/-/typescript-3.9.10.tgz: unexpected end of file".
```

in the log files. Maybe something with the network.

**Review by kaizencc — APPROVED** [review]

File: packages/@aws-cdk/aws-msk/test/cluster.test.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ describe('MSK Cluster', () => {
     [msk.KafkaVersion.V3_1_1, '3.1.1'],
     [msk.KafkaVersion.V3_2_0, '3.2.0'],
     [msk.KafkaVersion.V3_3_1, '3.3.1'],
+    [msk.KafkaVersion.V3_3_2, '3.3.2'],
+
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-msk/test/cluster.test.ts:44]:
```suggestion
```

**Review by kaizencc — APPROVED** [review]

**Pull request merged**